sw/device/silicon_creator/lib/boot_svc/boot_svc_msg.h

```diff
@@ -0,0 +1,41 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_BOOT_SVC_BOOT_SVC_MSG_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_BOOT_SVC_BOOT_SVC_MSG_H_
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/silicon_creator/lib/boot_svc/boot_svc_empty.h"
+#include "sw/device/silicon_creator/lib/boot_svc/boot_svc_header.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * A Boot Services message.
+ *
+ * This is defined as a union where the common initial sequence is a
+ * `boot_svc_header_t`. This makes it possible to store and read different types
+ * of messages to the same location without invoking undefined behavior.
+ */
+typedef union boot_svc_msg {
+  /**
+   * Common initial sequence.
+   */
+  boot_svc_header_t header;
+  /**
+   * Empty boot services message.
+   */
+  boot_svc_empty_t empty;
+} boot_svc_msg_t;
+// TODO: Add an assertion for checking that CHIP_BOOT_SVC_MSG_SIZE_MAX is
+// up to date after defining structs for other messages.
```

Comment:
Not yet, we want the size of the empty message to be equal to the size of the largest message (within the set of the remaining messages). I think I'll end up adding something like
```
sizeof(t1) == SIZE_MAX || sizeof(t2) == SIZE_MAX || ...
```

But this expression will not include `sizeof(boot_svc_empty_t)` since it must be equal to `SIZE_MAX` by definition.